drivers/flash/CMakeLists.txt

```diff
@@ -56,6 +56,13 @@ if(CONFIG_SOC_FLASH_STM32)
   endif()
 endif()
 
+if(CONFIG_SOC_FLASH_GD32)
+  zephyr_library_sources(flash_gd32.c)
+
+  zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_GD32F4XX flash_gd32f4xx.c)
+  zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_GD32F3X0 flash_gd32f3x0.c)
```

Comment:
Do you have this driver right now? Otherwise you should drop it:

```diff
-zephyr_library_sources_ifdef(CONFIG_SOC_SERIES_GD32F3X0 flash_gd32f3x0.c)
```

Reply:
done.